File: scratch/convert_to_webp.py
import os
from PIL import Image

img_dir = r"c:\OSPanel\domains\quete\pre-alpha0.1.3-antigravity-git\assets\img"
files = [
    "joystick.png",
    "slide1.png",
    "slide2.png",
    "slide3.png",
    "slide4.png",
    "slide5.png"
]

for filename in files:
    src_path = os.path.join(img_dir, filename)
    dst_path = os.path.join(img_dir, filename.replace(".png", ".webp"))
    
    if os.path.exists(src_path):
        try:
            with Image.open(src_path) as img:
                img.save(dst_path, "WEBP", quality=85)
                print(f"Successfully converted {filename} to WebP ({os.path.getsize(dst_path)} bytes)")
        except Exception as e:
            print(f"Error converting {filename}: {e}")
    else:
        print(f"File not found: {filename}")

File: scratch/convert_to_webp.php
<?php
$imgDir = dirname(__DIR__) . '/assets/img/';
$files = [
    'joystick.png',
    'slide1.png',
    'slide2.png',
    'slide3.png',
    'slide4.png',
    'slide5.png'
];

foreach ($files as $file) {
    $src = $imgDir . $file;
    $dst = $imgDir . str_replace('.png', '.webp', $file);
    
    if (file_exists($src)) {
        $image = @imagecreatefrompng($src);
        if ($image !== false) {
            imagepalettetotruecolor($image);
            imagealphasave($image, true);
            if (imagewebp($image, $dst, 85)) {
                echo "Converted $file to WebP (" . filesize($dst) . " bytes)\n";
            } else {
                echo "Failed to save WebP for $file\n";
            }
            imagedestroy($image);
        } else {
            echo "Failed to open $file\n";
        }
    } else {
        echo "File $file not found\n";
    }
}
